site/guides/sponsoring-gas/gas-sponsorship-eligibility.md

```diff
@@ -0,0 +1,119 @@
+
+::: code-group
+
+```ts [setup-alchemy-provider.ts]
+import { provider } from "./provider.ts";
+
+// Find your Gas Manager policy id at: // [!code focus:10]
+//dashboard.alchemy.com/gas-manager/policy/create
+const GAS_MANAGER_POLICY_ID = "YourGasManagerPolicyId";
+
+// Link the provider with the Gas Manager. This ensures user operations
+// sent with this provider get sponsorship from the Gas Manager.
+provider.withAlchemyGasManager({
+  policyId: GAS_MANAGER_POLICY_ID,
+});
+```
+
+<<< @/snippets/provider.ts
+
+:::
+
+Then, before you call `sendUserOperation` on the provider, you can use [`buildUserOperation`](/packages/aa-core/provider/buildUserOperation.md) to check if the user operation is eligible for gas sponsorship or not.
+
+Calling `buildUserOperation` method on the provider that is configured with the gas manager (`withAlchemyGasManager`) will run the middleware pipeline including the paymaster middleware, which will thrown an error if the user operation is not eligible for gas sponsorship under the policy.
+
+```ts [check-gas-sponsorship-eligibility.ts]
+let elligible = true;
+
+try {
+  await provider.buildUserOperation({
+    target: "0xTargetAddress",
+    data: "0xCallData",
+    value: 0n, // value in bigint or leave undefined
+  });
+} catch (e) {
+  // if the user operation is not eligible for gas sponsorship, error will be thrown and caught here
+  elligible = false;
+}
+
+console.log(
+  `User Operation is ${
+    elligible ? "elligible" : "inelligible"
+  } for gas sponsorship`
+);
+```
+
+## 2. How to Bypass the Paymaster Middleware For User Operations Not Eligible for Gas Sponsorship
+
+So how would you handle the user operation not elligible for gas sponsorship? You might want to unblock your users to be able to make the user operation transaction without gas sponsorship. This section of the guide explains how you can bypass the `PaymasterMiddleware` and still be able to send the user operation without gas sponsorship. This will let your users to make the transaction by paying the gas fee themselves.
+
+When you call `sendUserOperation` method on `SmartAccountProvider`, in addition to the `UserOperationCallData` or `BatchUserOperationCallData`, you can pass in an optional param `overrides` where you can specify override values for `maxFeePerGas`, `maxPriorityFeePerGas` or `paymasterAndData`. In order to bypass the `PaymasterMiddleware`, you can specify the `paymasterAndData` override. For example, if you set the `paymasterAndData` field in the overrides as `0x`, this will make the user operation to be sent where the gas fee is not paid by the paymaster but by the user's own smart account.
+
+```ts [bypass-paymaster-middleware.ts]
+// Baypass paymaster middleware by passing in the paymasterAndData override
+
+/*
+ * export type UserOperationOverrides = Partial<
+ *   Pick<
+ *     UserOperationStruct,
+ *     "maxFeePerGas" | "maxPriorityFeePerGas" | "paymasterAndData"
+ *   >
+ * >;
+ */
```

Comment:
let's just link to the `sendUserOperation` doc in the paragraph above, which will have this field's input types laid out.